repos/system_upgrade/el7toel8/actors/checkfstab/actor.py

```diff
@@ -0,0 +1,81 @@
+import os
+import time
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common.fstab import drop_xfs_options
+from leapp.libraries.stdlib import api
+from leapp.models import Report, FstabSignal
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp import reporting
+
+
+PREFACE = """# Below are the modified contents of /etc/fstab that have been generated for
+# the upgraded system. Changes to this file will not be reflected. If you
+# believe the file has been modified incorrectly, please report a bug and
+# attach this file and the current /etc/ftsab.
+
+# Generated by leapp on {}
+# ========================================
+"""
+FSTAB_LOGFILE = '/var/log/leapp/fstab.new'
+
+
+class CheckFstab(Actor):
+    """
+    Scan /etc/fstab for XFS mount options that have been removed in RHEL 8.
+
+    In case one or more of the options are present in /etc/fstab, generate new contents of the file
+    for the target system and provide it to the user via /var/log/leapp/fstab.new for inspection.
+    """
+
+    name = 'check_fstab'
+    consumes = ()
+    produces = (Report, FstabSignal)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        fstab_old = []
+        try:
+            with open('/etc/fstab') as f:
```

Comment:
you cannot work with files in the check phase. expecte logic is - get all required data about the fstab during the scanner phase. work with the data during the check phase.